frontend/src/services/QueryService.ts

```diff
@@ -19,4 +20,19 @@ export default abstract class QueryService {
     query?: LimitQuery,
     cancellation?: CancellationToken
   ): Promise<QueryResult>;
+
+  /**
+   * Searches through all entities and entity types and returns those, that
+   * match the searchString. Only returns those values with the prefix of searchStrings.
+   * Values with n spaces are considered as n+1 single values.
+   * @param searchString The string to search for.
+   * @param path the path of the search API describes what to search for, can be 'all' 'nodes' or edges'.
+   * @param cancellation A CancellationToken used to cancel the asynchronous operation.
+   * @returns A promise that when evaluated, contains a {@link SearchResult} (nodes, edges, edgetypes and nodetypes).
+   */
+  public abstract fullTextSearch(
```

Comment:
Why isn't this in a dedicated service? In the backend it is not part of this service and the HTTP API is on the subpath `search/` while all urls this service maes available is on the root path.

Reply:
Thanks for the quick review! Moved this to a dedicated service in ea9e726